### PR TITLE
Add swagger spec for Synapse Sql Gen3 resources

### PR DESCRIPTION
**Pull request by phkanekal**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ravbhatnagar** [comment]
please schedule a review meeting with ARMAPIReview and Azure API review board (azureapirbcore)

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/examples/PatchSqlDatabase.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "202": {},
+    "200": {
+      "body": {
+        "properties": {
+          "collation": "SQL_Latin1_General_CP1_CI_AS",
+          "maxSizeBytes": 263882790666240,
+          "databaseGuid": "00000000-0000-0000-0000-000000000004",
+          "creationDate": "2020-06-03T04:05:06Z"
+        },
+        "location": "inprocess",
```
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/examples/PatchSqlDatabase.json:20]: locations in examples should be real locations

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the specified SqlDatabase.",
+            "schema": {
+              "$ref": "#/definitions/SqlDatabase"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 404 ServerNotInSubscriptionResourceGroup - Specified server does not exist in the specified resource group and subscription.\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 ResourceNotFound - The requested resource was not found."
```
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json:51]: The default response should return the error schema defined in your types.json

Comment (pilor) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json:51]: applies all over

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+          },
+          "202": {
```
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json:102]: 202 should not be used for async PUT.  See https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creatingupdating-using-put

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+          "description": "The max size of the database expressed in bytes.",
+          "type": "integer"
+        },
+        "databaseGuid": {
+          "format": "uuid",
+          "description": "The Guid of the database.",
+          "type": "string",
+          "readOnly": true
+        },
+        "creationDate": {
```
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json:242]: Creation/LastModified data should be pulled up to the top level using the common systemdata contract.  See https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#system-metadata-for-all-azure-resources.  There is a common definition for systemdata in the top level types.json

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/sqlDatabases/{sqlDatabaseName}": {
```
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json:19]: You are missing a collection GET API that lists all sqlDatabases in the workspace

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlPool.json [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+          },
+          "202": {
```
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlPool.json:102]: 202 is not used for PUTs

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+        "x-ms-examples": {
+          "Updates a sqldatabase.": {
+            "$ref": "./examples/PatchSqlDatabase.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SqlDatabaseProperties": {
```
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json:219]: if these are created asynchronously they should have a "provisioningState" property in their property bag.  See the async PUT link I pasted above

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlPool.json [hunk 8]
```diff
@@ -0,0 +1,341 @@
...
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SqlPoolProperties": {
+      "description": "The sql pool's properties.",
+      "type": "object",
+      "properties": {
+        "version": {
```
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlPool.json:223]: pull this up to "kind" if you can.  I.e.  "poolv2" and "poolv3"

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlPool.json [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+            "modelAsString": true
+          }
+        },
+        "sqlPoolGuid": {
+          "format": "uuid",
+          "description": "The Guid of the sql pool.",
+          "type": "string",
+          "readOnly": true
+        },
+        "creationDate": {
```
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlPool.json:261]: same "systemdata" contract comment as sqlDatabases

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlPool.json [hunk 10]
```diff
@@ -0,0 +1,341 @@
...
+          "type": "string",
+          "readOnly": true
+        },
+        "requestedServiceObjectiveName": {
+          "description": "The requested service level objective name of the sql pool.",
+          "type": "string",
+          "readOnly": true
+        },
+        "currentSku": {
+          "$ref": "../../../../common/v1/types.json#/definitions/Sku",
```
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlPool.json:278]: why does SKU need to be at the top level of the resource and here?

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlPool.json [hunk 11]
```diff
@@ -0,0 +1,341 @@
...
+        "x-ms-examples": {
+          "Updates a sqlpool.": {
+            "$ref": "./examples/PatchSqlPool.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SqlPoolProperties": {
```
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlPool.json:219]: this also needs provisioningState for async put

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json (hunk 3 above)
Comment (phkanekal) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2020-04-01-preview/sqlDatabase.json:102]: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#202-accepted-and-location-headers

in our scenario if the PUT fails, we actually drop the resource. this is in line with what Microsoft.Sql RP does as well so after talking to them, they suggested to keep this 202 pattern as is.

**Review by pilor — APPROVED** [review]

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**